Add support for acsystem and multi rs devices

blammit opened the pull request.

DanielMcInnes commented on pages/settings/devicelist/DeviceListPage.qml:
```diff
@@ -38,15 +44,7 @@ Page {
-			params = { "bindPrefix" : device.serviceUid }
-			summary = [ Global.system.systemStateToText(device.state) ]
-			break;
+			return null
```
maybe just remove the "multi" case and let it fall through to the default?

blammit replied:
Added a comment here.

DanielMcInnes commented:
When the mode is 'On', and I click the 'Mode' button, the radio button selector pops up. I think the 'On' button should already be selected, but it isn't. 

![Screenshot from 2024-06-04 13-22-04](https://github.com/victronenergy/gui-v2/assets/6416318/b17658c9-7def-44d4-a6cb-9d1066c07ec0)

jpetrell commented on pages/vebusdevice/PageVeBus.qml:
```diff
@@ -379,11 +342,11 @@ Page {
-			outputPhases: acOutput.phases
-			acActiveInputPower: _acActiveInputPower
-			acOutputPower: _acOutputPower
+			phaseCount: _numberOfPhases.value || 0
```
gui-v2 has 57 nrOf* properties, 90 numberOf* properties and 691 *count properties. Count is winning. :smile:

jpetrell commented:
> Specify the phase and total uids to avoid creating unnecessary rows and VeQuickItems for unused phaes

Typo "phaes" in one of the commit messages. ^^

jpetrell commented on pages/settings/devicelist/rs/PageMultiRs.qml:
```diff
@@ -0,0 +1,265 @@
+			}
+
+			ListNavigationItem {
+				text: CommonWords.history
```
Yeah if you would have only one history item than "History" would be enough, but there is "Overall history" also below.

blammit replied:
Agreed, fixed to be "Daily history".

This doesn't directly map to the gui-v1 settings here, as gui-v2 can show the solar history that combines the tracker daily history with the main daily history.

DanielMcInnes replied:
When I click on daily history, and view the chart for the last 30 days, the chart is empty. Is this just because the target installation doesn't have the latest firmware? I'm looking at the 'J Cal' installation

blammit replied:
Just had a look, and that installation has zero yield over the last few days, and it only has two days of history, so that's why no bars are visible. Perhaps we should disable the 7 days / 30 days if there are no more than 2 days of history.

jpetrell commented on components/InverterModeDialogLauncher.qml:
```diff
@@ -38,6 +42,7 @@ QtObject {
 	property Component _inverterChargerModeDialogComponent: Component {
 		InverterChargerModeDialog {
 			isMulti: root.isMulti
+			hasPassthroughSupport: _hasPassthroughSupport.value === 1
```
Should hasPassthroughSupport be defined for InverterChargerModeDialog in VeBusDeviceModeButton.qml?

blammit replied:
Thanks, good catch, fixed.

jpetrell commented on pages/settings/devicelist/rs/PageRsSystem.qml:
```diff
@@ -0,0 +1,208 @@
+					if (currentLimitIsAdjustable.isValid && currentLimitIsAdjustable.value) {
+						Global.dialogLayer.open(currentLimitDialogComponent, { value: currentLimit.value })
+					} else {
+						Global.showToastNotification(VenusOS.Notification_Info, qsTrId("settings_rs_current_limit_not_adjustable"), 5000)
```
No Engineering English translation defined?

blammit replied:
Updated to use same text as gui-v1.

jpetrell commented:
Overall LGTM, but would like somebody else to approve also.

blammit commented:
> When the mode is 'On', and I click the 'Mode' button, the radio button selector pops up. I think the 'On' button should already be selected, but it isn't.

Thanks! Fixed.

blammit commented on pages/settings/devicelist/rs/PageRsSystem.qml:
```diff
@@ -0,0 +1,208 @@
+								readonly property VeQuickItem _customName: VeQuickItem { uid: rsDeviceDelegate.rsServiceUid ? rsDeviceDelegate.rsServiceUid + "/CustomName" : "" }
+								readonly property VeQuickItem _productName: VeQuickItem { uid: rsDeviceDelegate.rsServiceUid ? rsDeviceDelegate.rsServiceUid + "/ProductName" : "" }
+								readonly property VeQuickItem _state: VeQuickItem { uid: rsDeviceDelegate.rsServiceUid ? rsDeviceDelegate.rsServiceUid + "/State" : "" }
+
```
True, moved to separate component.

blammit commented:
While testing this PR I noticed some solar history bugs, so I've fixed them as part of this PR so that it can be tested together. See https://github.com/victronenergy/gui-v2/issues/1188.

Also updated the behavior when the Inverter/charger widget is clicked (as described here https://github.com/victronenergy/gui-v2/issues/1077#issuecomment-2148676154) as discussed in the meeting yesterday. This is done in the top commit (96a69e70aa7a31c5f5f1d22f021e0aac165a6fa2).

DanielMcInnes reviewed (approved):
looks great

chriadam commented on pages/settings/devicelist/rs/PageMultiRs.qml:
```diff
@@ -0,0 +1,269 @@
+		id: singleTrackerComponent
+
+		ListQuantityGroup {
+			function current() {
```
should it be a property?

chriadam commented on pages/settings/devicelist/rs/PageRsSystemDevices.qml:
```diff
@@ -0,0 +1,77 @@
+			id: rsDeviceDelegate
+
+			required property BaseDevice device
+
```
nit: extra whitespace ;-)

chriadam commented on data/common/SolarTrackerDailyHistory.qml:
```diff
@@ -0,0 +1,33 @@
+/*
+** Copyright (C) 2023 Victron Energy B.V.
```
nit: 2024

chriadam commented on data/common/SolarTrackerDailyHistory.qml:
```diff
@@ -0,0 +1,33 @@
+	// Prefix is: com.victronenergy.solarcharger.tty0/History/Daily/<day>/Pv/<pv-index>
+	property string uidPrefix
+
+	readonly property real yieldKwh: _yieldKwh.value == undefined ? NaN : _yieldKwh.value
```
maybe isValid checks instead?

chriadam reviewed (approved):
Minor nits, but LGTM.  Nice work.
